GitUI/Editor/FileViewer.cs

```diff
@@ -591,64 +591,64 @@ public async Task ViewPatchAsync(string fileName, string text, Action openWithDi
                 {
                     if (isText)
                     {
-                        // A submodule 'patch' is handled as text
-                        ResetForText(fileName);
-                        internalFileViewer.SetText(text, openWithDifftool, isDiff: false);
-                    }
-                    else
-                    {
-                        ResetForDiff(fileName);
-                        internalFileViewer.SetText(text, openWithDifftool, isDiff: true);
+                        // A submodule 'patch', a new file or patch could not be evaluated
+                        return ViewText(fileName, text, openWithDifftool);
                     }
 
+                    ResetForDiff(fileName);
+                    internalFileViewer.SetText(text, openWithDifftool, isDiff: true);
+
                     TextLoaded?.Invoke(this, null);
                     return Task.CompletedTask;
```

Comment:
Early exit misses `TextLoaded?.Invoke(this, null);`

Reply:
Applying to #7926

Reply:
This is actually OK, Invoke is called in ViewText(), so no change

Reply:
Made a change to move this logic to the extension
https://github.com/gitextensions/gitextensions/pull/7926#issuecomment-609995516